Makefile

```diff
@@ -134,6 +134,14 @@ test: bundle generate lint envtest helm-lint ginkgo ## Run tests.
 	-rm -f cover.out
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" $(GINKGO) $(TEST_ARGS) $(TEST_PACKAGES)
 
+# Example build target showing how to run unit tests, excluding syncthing from being built entirely
+# this is not bothering to exclude syncthing during linting but should be possible
+.PHONY: test-disable-syncthing
+TEST_ARGS_DISABLE_SYNCTHING ?= --tags disable_syncthing
+test-disable-syncthing: bundle generate lint envtest helm-lint ginkgo ## Run tests.
+	-rm -f cover.out
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" $(GINKGO) $(TEST_ARGS) $(TEST_ARGS_DISABLE_SYNCTHING) $(TEST_PACKAGES)
+
```

Comment:
I'm wondering whether we can make this more general. I know it's just an example, but here's what I'm thinking...

Usage:
```console
$ TAGS=disable_syncthing make <target(s)>
```

We could then have at the top of the Makefile something that picks up whether TAGS is defined and sets secondary variables for build flags for go and for the container build arg...
```makefile
GOBUILDTAGS:=
CONTANIERBUILDTAGS:=
ifdef TAGS
    GOBUILDTAGS:=-tags $TAGS
    CONTAINERBUILDTAGS:=--build-arg "tags_arg=$TAGS"
endif
```
... then we just always add GOBUILDTAGS, CONTAINERBUILDTAGS at the appropriate places

The syntax probably has errors, but it's the general idea... thoughts?

Reply:
I like this idea, one thing is if someone does build with a mover disabled, this won't disable actually packaging the mover itself (such as syncthing for example) - it would just disable the syncthing mover code in the volsync manager binary.  Do you think it's worth looking at disabling the mover packaging in the makefile too?  I guess I was hoping the example make targets are showing some work might be needed, but here's a reference for later if we get there.

Another current limitation is unit tests won't all run well if we disable certain movers - rsync is required for the controller tests (wonder if I should change these to use rsync-tls actually), and we have some rclone mover tests in controllers/ as well so currently there's some setup in suite_test.go that requires rclone.